.github/workflows/tests-smoke.yaml

```diff
@@ -187,7 +187,7 @@ jobs:
           cilium_pod=$(kubectl -n kube-system get po -o name --field-selector=status.phase==Running -l 'k8s-app=cilium' -o jsonpath='{.items[0].metadata.name}' )
           kubectl -n kube-system exec $cilium_pod -- sh -c "apt update && apt install curl -y"
           kubectl -n kube-system exec $cilium_pod -- curl http://localhost:9090/metrics > metrics.prom
-          GO111MODULE=on go get github.com/prometheus/prometheus/cmd/promtool@e4487274853c587717006eeda8804e597d120340 # This is commit hash for v2.24.1
+          go install github.com/prometheus/prometheus/cmd/promtool@v2.31.1
```

Comment:
Why `2.31.1`? Perhaps we could update to `v2.34.0` and use the pinned commit `881111fec4332c33094a6fb2680c71fffc427275`?

Reply:
This is updated to v2.34.0 as part of the switch to the binary release in the second to last commit. I'll keep the existing version in this commit.